### PR TITLE
[Doc] extensibility framework overview

### PR DESCRIPTION
**Pull request by jcalcaben**
## Description

This PR creates an overview topic for the Extensibility Framework based on the draft provided by @zetlen.

## Related Issue

Closes PWA-800

## Acceptance

Any developer familiar with the framework implementation

### Verification Stakeholders

Any developer familiar with the framework implementation

### Verification Steps

1. Navigate to the `pwa-devdocs` directory: `cd pwa-devdocs`
1. Run the HTML preview server: `yarn develop`
1. Verify site builds
1. Navigate to `/pwa-buildpack/extensibility-framework/`
1. Verify correct page loads
1. Verify **Extensibility framework** shows up on the left navigation
1. Verify page content is accurate

## Checklist

* I have added tests to cover my changes, if necessary.
* I have added translations for new strings, if necessary.
* I have updated the documentation accordingly, if necessary.

### CODE REVIEW

File: pwa-devdocs/src/pwa-buildpack/extensibility-framework/index.md [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+
+![extensibility-overview][]
+
+This is different from a _plugin architecture_ where the application detects and dispatches plugins as the front end loads in the browser.
+The more plugins you install with this architecture, the slower the application gets as it becomes bloated with overhead processes.
+
+### Interceptor pattern
+
+The extensibility framework implements an [interceptor pattern][] to allow changes to the build process.
+The interceptor pattern lets a module **C** intercept the flow of logic between module **A** and module **B** and add it's own logic.
```
Comment (supernova-at) [inline pwa-devdocs/src/pwa-buildpack/extensibility-framework/index.md:31]:
```suggestion
The interceptor pattern lets a module **C** intercept the flow of logic between module **A** and module **B** and add its own logic.
```

File: pwa-devdocs/src/pwa-buildpack/extensibility-framework/index.md [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+            componentName: 'PageBuilder',
+            importPath: '@magento/pagebuilder'
+        });
+    });
+```
+
+## Declare files
+
+A _declare file_ lists the Targets available for interception in an extension or package.
+During build time, the framework looks for this file using the value set for `pwa-studio.targets.intercept` in your project's `package.json` file.
```
Comment (supernova-at) [inline pwa-devdocs/src/pwa-buildpack/extensibility-framework/index.md:83]: Should this be

```suggestion
During build time, the framework looks for this file using the value set for `pwa-studio.targets.declare` in your project's `package.json` file.
```

?

File: pwa-devdocs/src/pwa-buildpack/extensibility-framework/index.md [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+module.exports = targets => {
+  targets.declare({
+    perfReport: new targets.types.AsyncParallel(['report'])
+    socialIcons: new targets.types.SyncWaterfall(['iconlist'])
+  });
+}
+```
+
+The example provided declares two Targets that this project or other modules can intercept, a `perfReport` Target and a `socialIcons` Target.
+Intercept files can access these Targets using `targets.of(<package name>).perReport` and `targets.of(<package name>).socialIcons` respectively.
```
Comment (supernova-at) [inline pwa-devdocs/src/pwa-buildpack/extensibility-framework/index.md:115]:
```suggestion
Intercept files can access these Targets using `targets.of(<package name>).perfReport` and `targets.of(<package name>).socialIcons` respectively.
```

File: pwa-devdocs/src/pwa-buildpack/extensibility-framework/index.md [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+
+The _intercept file_ describes the targets you want to intercept and define or extend.
+During build time, the framework looks for this file using the value for `pwa-studio.targets.intercept` in your project's `package.json` file.
+These files execute after all the declare files register their Targets.
+
+Your intercept files must export a function that accepts a TargetProvider object as a parameter.
+This object gives you access to all available Targets in your project that let you make customizations, gather data, change the build itself, or develop and call your own declared targets.
+The TargetProvider object provides an `of()` function and an `own` property to access Targets in other packages or Targets within its own module respectively.
+
+For more information on how intercept files work, see the tutorial on how to [Intercept a Target][].
```
Comment (supernova-at) [inline pwa-devdocs/src/pwa-buildpack/extensibility-framework/index.md:135]: Even a simple example here would be great

**Review by supernova-at — CHANGES_REQUESTED** [review]
This is great stuff!

Just some very minor change requests

File: pwa-devdocs/src/pwa-buildpack/extensibility-framework/index.md (hunk 4 above)
Comment (jcalcaben) [reply pwa-devdocs/src/pwa-buildpack/extensibility-framework/index.md:135]: Added in 84ef809

**Review by supernova-at — APPROVED** [review]

**Pull request merged**